packages/dd-trace/src/appsec/callbacks/ddwaf.js

```diff
@@ -0,0 +1,142 @@
+'use strict'
+
+const log = require('../../log')
+const Addresses = require('../addresses')
+const Gateway = require('../../gateway/engine')
+const Reporter = require('../reporter')
+
+let warned = false
+
+const validAddressSet = new Set(Object.values(Addresses))
+
+const DEFAULT_MAX_BUDGET = 5e3 // µs
+
+// TODO: put reusable code in a base class
+class WAFCallback {
+  static loadDDWAF (rules) {
+    try {
+      // require in `try/catch` because this can throw at require time
+      const { DDWAF } = require('@datadog/native-appsec')
+
+      return new DDWAF(rules)
+    } catch (err) {
+      if (!warned) {
+        log.warning('AppSec could not load native package. In-app WAF features will not be available.')
+        warned = true
+      }
+
+      throw err
+    }
+  }
+
+  constructor (rules) {
+    this.ddwaf = WAFCallback.loadDDWAF(rules)
+    this.wafContextCache = new WeakMap()
+    this.ruleNames = new Map()
+
+    // closures are faster than binds
+    const self = this
+    const method = (params, store) => {
+      self.action(params, store)
+    }
+
+    // might be its own class with more info later
+    const callback = { method }
+
+    this.subscriptions = []
+    const subscriptionGroups = new Set()
+
+    for (const rule of rules.events) {
+      this.ruleNames.set(rule.id, rule.name)
+
+      for (const condition of rule.conditions) {
+        let addresses = condition.parameters.inputs.map((address) => address.split(':', 2)[0])
+
+        if (!addresses.every((address) => validAddressSet.has(address))) {
```

Comment:
This is not a reason for invalidity of rule, let's remove the log

Reply:
You mean just ignoring the rule without logging ?